plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy

```diff
@@ -711,77 +711,85 @@ class AzBatchService implements Closeable {
     }
 
     protected void createPool(AzVmPoolSpec spec) {
+        try {
```

Comment:
Generally speaking, when a long function needs to be wrapped in a try-catch like this, we like to put the try-catch in a separate wrapper function:

```groovy
protected void safeCreatePool(AzVmPoolSpec spec) {
    try {
        createPool(spec)
    }
    catch( HttpResponseException e ) {
        // ...
    }
}
```